src/SWP/Component/GeoIP/Checker/GeoIPChecker.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Superdesk Publisher Geo IP Component.
+ *
+ * Copyright 2019 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2019 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Component\GeoIP\Checker;
+
+use GeoIp2\Exception\AddressNotFoundException;
+use SWP\Component\GeoIP\Model\Place;
+use SWP\Component\GeoIP\Reader\ReaderInterface;
+
+class GeoIPChecker
+{
+    /** @var ReaderInterface */
+    private $reader;
+
+    public function __construct(ReaderInterface $reader)
+    {
+        $this->reader = $reader;
+    }
+
+    public function isGranted(string $ipAddress, Place $place): bool
```

Comment:
i don't get it. `isGranted` returns `false` when state, or country matches. And in other cases returns `true`?

Reply:
yes, because if the article country equals to the country based on IP it should not allow seeing the article.

Reply:
But in places we should provide allowed countries/states - not excluded.

Reply:
we agreed that when the place in SD will be defined, readers from these "places" will not be able to read the article.